### PR TITLE
feat(send): send tokens functionality v1

### PR DESCRIPTION
**Pull request by mateodaza**
Only xDAI for now.
Please be sure to fill your wallet with some xDAI before trying, removed mock tokens from there

Next version will have ERC20s integrated, function is ready for that

### CODE REVIEW

**Review by yussdev — APPROVED** [review]

File: src/components/SendModal/index.tsx [hunk 1]
```diff
@@ -8,78 +10,200 @@ import { ModalHeader } from '../ModalHeader'
...
+        text && setDestination(text)
+      } catch (error) {
+        console.error('Failed to read clipboard contents: ', error)
+      }
+    }
+
+    const ethToken: ICryptoToken = {
+      name: 'xDAI', /// TODO: check on current network
+      value: Number(ethBalance),
+      price: 3000,
```
Comment (jainkrati) [inline src/components/SendModal/index.tsx:69]: lets add a todo here

File: src/components/SendModal/index.tsx [hunk 2]
```diff
@@ -8,78 +10,200 @@ import { ModalHeader } from '../ModalHeader'
...
+            <Typography color="green">{successMessage}</Typography>
+          ) : (
+            amountError && <Typography color="red">{amountError}</Typography>
+          )}
+          <Button
+            onClick={async () => {
+              setDestinationWarning(null)
+              setDestinationError(null)
+              setAmountError(null)
+              if (!destination) {
```
Comment (jainkrati) [inline src/components/SendModal/index.tsx:182]: good to see validations in place :)

File: src/hooks/useSafeAuth.ts [hunk 3]
```diff
@@ -21,6 +24,63 @@ export const useSafeAuth = () => {
...
+        const amountInWei = ethers.parseUnits(amount, 'ether') // Assuming token has 18 decimals
+        const txResponse = await tokenContract.transfer(toAddress, amountInWei)
+        await txResponse.wait()
+        console.log(
+          `ERC20 token transfer successful with hash: ${txResponse.hash}`
+        )
+        return txResponse.hash
+      }
+    } catch (error) {
+      console.error('Transaction failed:', error)
```
Comment (jainkrati) [inline src/hooks/useSafeAuth.ts:80]: we wil need to show a popup for same in UX; can be doe in next PR

**Review by jainkrati — APPROVED** [review]
Approving for the demo. Lets take PR comments in next PR

**Pull request merged**